packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -436,23 +439,23 @@ class _HelperErrorState extends State<_HelperError> with SingleTickerProviderSta
 
     if (_controller.isCompleted) {
       _helper = null;
-      if (widget.errorText != null) {
+      if (widget.errorText != null || widget.error != null) {
         return _error = _buildError();
       } else {
         _error = null;
         return empty;
       }
     }
 
-    if (_helper == null && widget.errorText != null) {
+    if (_helper == null && _error != null) {
```

Comment:
I think this should be `if (_helper == null && (widget.errorText != null || widget.error != null))`.

Reply:
Okay